gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/OrderServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+    private GiftCertificateService giftCertificateService;
+    private List<Order> orderList;
+    private CustomPagination pagination;
+
+    @BeforeAll
+    void beforeAll() {
+        orderDao = mock(OrderDaoImpl.class);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+        userService = mock(UserServiceImpl.class);
+        giftCertificateService = mock(GiftCertificateServiceImpl.class);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+
+        user = new User();
+        user.setId(USER_ID);
+        user.setFirstName(FIST_NAME);
+        user.setLastName(LAST_NAME);
+        user.setLogin(LOGIN);
+
+        Tag tag = new Tag();
+        tag.setId(TAG_ID);
+        tag.setName(TAG_NAME);
+        List<Tag> tagList = new ArrayList<>();
+        tagList.add(tag);
+
+        giftCertificate = new GiftCertificate();
+        giftCertificate.setId(GIFT_CERTIFICATE_ID);
+        giftCertificate.setName(NAME);
+        giftCertificate.setDescription(DESCRIPTION);
+        giftCertificate.setPrice(PRICE);
+        giftCertificate.setDuration(DURATION);
+        giftCertificate.setTagList(tagList);
+
+        order = new Order();
+        order.setId(ID);
+        order.setUser(user);
+        order.setGiftCertificate(giftCertificate);
+        order.setPrice(giftCertificate.getPrice());
+        order.setCreateDate(LocalDateTime.now());
+
+        orderDto = new OrderDto();
+        orderDto.setUserId(USER_ID);
+        orderDto.setGiftCertificateId(GIFT_CERTIFICATE_ID);
+
+        orderList = new ArrayList<>();
+        orderList.add(order);
+
+        pagination = new CustomPagination();
+        pagination.setPage(PAGE);
+        pagination.setSize(SIZE);
+
+        orderService = new OrderServiceImpl(orderDao, userService, giftCertificateService, paginationValidator);
+    }
+
+    @Test
+    void testFindAll() {
+        List<Order> expectedResult = orderList;
+        when(orderDao.findAll(pagination, Order.class)).thenReturn(orderList);
+        when(orderDao.findEntitiesNumber(Order.class)).thenReturn(ENTITIES_NUMBER);
+        when(paginationValidator.validatePagination(pagination, ENTITIES_NUMBER)).thenReturn(pagination);
+        List<Order> actualResult = orderService.findAll(pagination);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindById() {
+        Order expectedResult = order;
+        when(orderDao.findById(ID, Order.class)).thenReturn(Optional.ofNullable(order));
+        Order actualResult = orderService.findById(ID);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindByIdShouldTrowOrderNotFoundException() {
+        when(orderDao.findById(ID, Order.class)).thenReturn(Optional.empty());
+        assertThrows(OrderNotFoundException.class, () -> orderService.findById(ID));
+    }
+
+    @Test
+    void testFindByUserId() {
+        List<Order> expectedResult = orderList;
```

Comment:
I suppose you can directly use orderList and do not create an additional variable